lib/matplotlib/backends/backend_pgf.py

```diff
@@ -834,7 +834,7 @@ def print_pgf(self, fname_or_fh, *args, **kwargs):
 
         # figure out where the pgf is to be written to
         if isinstance(fname_or_fh, str):
-            with codecs.open(fname_or_fh, "w", encoding="utf-8") as fh:
+            with open(fname_or_fh, "w", encoding="utf-8") as fh:
```

Comment:
As far as I understand, this is not identical, since `codecs.open` always uses binary mode.

Reply:
The *only* difference from binary mode (given that we're giving an explicit encoding) (see the Note under https://docs.python.org/3/library/functions.html#open) is that there's no newline conversion, i.e. previously this would generate files with `\n` newlines on Windows and Linux but now this generates files with `\r\n` newlines on Windows and `\n` on Linux.

Which, if anything, is more correct IMO (for example, notepad (dunno who uses that to edit tex files on Windows, but I guess that's not the only tool) doesn't like files that use \n as newline...).

Reply:
Yes the newline is the difference. I'd say, if it ain't broken don't fix it - you can work perfectly fine with `\n`-only files in windows. On the other hand, if you want that change, it's fine by me. We have just to be sure that all the (latex?) tools that are running on that file will handle `\r\n` properly.

Side remark: [Notepad just got support for unix line endings](https://tech.slashdot.org/story/18/05/08/2149216/windows-notepad-finally-supports-unix-mac-os-line-endings) (not that there are still a lot of reasons to use other editors on windows which already support unix line endings for decades).

Reply:
There's apoproximately a million other places where we generate files with \r\n endings on Windows, I don't see why pgf would be different.  Unless you want to ban the use of open() in favor of codecs.open throughout the codebase...

Reply:
It's fine. I just wanted to make sure, we're making this change consciously.